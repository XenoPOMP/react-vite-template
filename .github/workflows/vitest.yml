name: 'Testing'
on:
  push:
  pull_request:

jobs:
  test:
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
      - uses: actions/checkout@v2
      - name: 'Install Node'
        uses: actions/setup-node@v2
        with:
          node-version: '16.x'
      - name: 'Install Deps'
        run: npm i yarn -g && yarn
      - name: 'Test'
        run: yarn coverage
      - name: 'Report Coverage'
        if: always() # Also generate the report if tests are failing
        uses: davelosert/vitest-coverage-report-action@v2
