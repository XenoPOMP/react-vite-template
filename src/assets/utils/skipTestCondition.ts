import useEnv, { TestingMode } from '@hooks/useEnv';

const { TESTING_MODE } = useEnv();

/**
 * Determines skip test condition.
 * @param {TestingMode} mode       target mode
 * @returns {boolean}              false if current mode isn`t the same as target one
 *                                 or current mode is FULLSTACK
 */
const skipTestCondition = (mode: TestingMode): boolean => {
  if (TESTING_MODE === 'FULLSTACK') {
    return false;
  }

  return TESTING_MODE !== mode;
};

export { skipTestCondition as default };
