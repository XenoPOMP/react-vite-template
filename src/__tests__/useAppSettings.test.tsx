import { fireEvent } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import skipTestCondition from '@utils/skipTestCondition';

import { defaultTestingPageRender } from './defaults';

/**
 * Test useAppSettings hook.
 */

describe.skipIf(skipTestCondition('FRONTEND'))('useAppSettings hook', () => {
  test('Match snapshot', () => {
    expect(defaultTestingPageRender()).toMatchSnapshot();
  });

  test('buttons` callback work right', () => {
    const page = defaultTestingPageRender().asFragment();

    const langText = page.querySelector('#lang-output');
    const buttons = page.querySelectorAll('button.lang-switch');

    /** Click each button. */
    buttons.forEach(button => fireEvent.click(button));

    expect(1);
  });
});
