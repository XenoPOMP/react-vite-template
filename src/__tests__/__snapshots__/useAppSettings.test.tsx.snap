// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useAppSettings hook > Match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          App version: 
          0.0.0
        </p>
        <p
          id="lang-output"
        >
          Language: 
          en
        </p>
        <button
          class="lang-switch"
        >
          Change lang to Russian
        </button>
        <button
          class="lang-switch"
        >
          Change lang to English
        </button>
        <p
          id="check-preview"
        >
          false
        </p>
        <button
          id="checkbox-test"
        >
          <label
            class="_checkBoxLabel_05fc2d"
          >
            <input
              class="_boxInput_05fc2d"
              type="checkbox"
            />
            <span
              class="_mark_05fc2d"
            >
              <svg
                fill="none"
                viewBox="0 0 18 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="15"
                  rx="2.5"
                  stroke="#737674"
                  width="15"
                  x="1.5"
                  y="1.5"
                />
                <rect
                  class="_active_05fc2d"
                  height="15"
                  rx="2.5"
                  stroke="#0066FF"
                  width="15"
                  x="1.5"
                  y="1.5"
                />
              </svg>
            </span>
          </label>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        App version: 
        0.0.0
      </p>
      <p
        id="lang-output"
      >
        Language: 
        en
      </p>
      <button
        class="lang-switch"
      >
        Change lang to Russian
      </button>
      <button
        class="lang-switch"
      >
        Change lang to English
      </button>
      <p
        id="check-preview"
      >
        false
      </p>
      <button
        id="checkbox-test"
      >
        <label
          class="_checkBoxLabel_05fc2d"
        >
          <input
            class="_boxInput_05fc2d"
            type="checkbox"
          />
          <span
            class="_mark_05fc2d"
          >
            <svg
              fill="none"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="15"
                rx="2.5"
                stroke="#737674"
                width="15"
                x="1.5"
                y="1.5"
              />
              <rect
                class="_active_05fc2d"
                height="15"
                rx="2.5"
                stroke="#0066FF"
                width="15"
                x="1.5"
                y="1.5"
              />
            </svg>
          </span>
        </label>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
